Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,362 @@
+    where :php:`$GLOBALS['TSFE']` is used. In several (but not all) cases, there
+    are alternatives available.
+
+From the source:
+
+    When calling a Frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the core development strives to get
+    rid of this in the future.
+
+:php:`$GLOBALS['TSFE']` is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or cli.
+
+Initializing :php:`$GLOBALS['TSFE']` in the Backend is sometimes done in code
+examples found online. TSFE is not initialized in the backend context by the
+core (and there is usually no need to do this).
+
+From the PHP documentation:
+
+    As of PHP 8.1.0, $GLOBALS is now a read-only copy of the global symbol table.
+    That is, global variables cannot be modified via its copy.
+
+https://www.php.net/manual/en/reserved.variables.globals.php
+
+Howtos
+======
+
+.. add howtos for common use cases, remove the not recommended, deprecated
+.. way if there is a better way
+
+Following are some examples which use TSFE and alternatives to using TSFE,
+where available:
+
+-------------
+
+Access the :php:`ContentObjectRenderer` (often referred to as "cObj"):
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->cObj
+
+Can be done as follows in Extbase controllers:
+
+.. code-block:: php
+
+    $contentObj = $this->configurationManager->getContentObject();
+
+-------------
+
+Access the current page id:
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->id
+
+Can be done using the ContentObjectRenderer (see previous example):
+
+.. code-block:: php
+
+    $pageId = $contentObj->data['pid'];
+
+-------------
+
+Access language settings is now done using the
+:ref:`language aspect <context_api_aspects_language>`.
+
+Get the request language of the current page as integer:
+
+.. code-block:: php
+
+    $context->getPropertyFromAspect('language', 'id');
+
+--------------
+
+Accessing information about Frontend and Backend users.
+
+.. code-block:: php
+
+    $feUser = $GLOBALS['TSFE']->fe_user;
+
+Some information via frontend and backend users con be obtained via the
+:ref: `user aspect <context_api_aspects_user>`. For example:
+
+.. code-block:: php
+
+    // return whether a frontend user is logged in
+    $context->getPropertyFromAspect('frontend.user', 'isLoggedIn');
+
+------------------
+
+To get the base URL of the current site, use site configuration.
+
+For example, to get the site via the current page id:
+
+.. code-block:: php
+
+    $siteFinder = GeneralUtility::makeInstance(SiteFinder::class);
+    getSiteByPageId($pageId);
+
+
+Hooks and events
+================
+
+.. todo::
+
+    Add overview of hooks and events that are dispatched / called in TSFE
+
+Properties and functions
+========================
+
+.. todo::
+    here some still used properties and functions can be listed and explained,
+    e.g. fe_user, id etc.
+
+
+Deprecated functionality (PHP)
```

Comment:
Do we really need this part? And if, please only add those which are available in v12 and maybe v11.

Reply:
The reason is the information is currently spread over several changelogs. You could also look in the source code, but then you don't have the context and the alternatives. 

The section is meant to be removed entirely in the future when there is less activity in that area. The reason I also added the older ones - we did not have this page in the past. I find it helpful to have one overview. (I realize we usually do not add notes if somethings was deprecated several versions ago. This is an exception because the page now only exists in main, and maybe 11.5).

The other reason - one motivation was to "catch" search engine results. If you search for one of the deprecated options, you are directed to pages not on docs.typo3.org or to the changelogs. I would like to direct people to this main page where the information is not fragmented and where they can find information for the specific TYPO3 version in the future (and then use the version selector if they were directed to an older version).

I find this especially important - to direct people to main docs pages - where there is lots of activity and changes and possibly outdated snippets exist in multiple places.

Reply:
What about this: For this PR we leave them as-is, merge it back to 11.5 and 10.4 and then clean up main (v12) for the outdated ones not related to v12 in a follow-up PR. This way we don't clutter up the recent documentation.

Additionally, we should not merge to 11.5 and 10.4 information for newer versions.

Reply:
Chris, that sounds good. Removing the v12 changelogs from the 11 and 10 backports should not be too much work, also removing the older changelogs from the main).

Reply:
I removed the deprecated section and the empty sections for now. It is a bit much for one change and reviewing for other branches as well. That could be added as followup patch.